README.md

```diff
@@ -23,6 +23,7 @@ This Action commits the contents of your Git tag to the WordPress.org plugin rep
 * `VERSION` - defaults to the tag name; do not recommend setting this except for testing purposes.
 * `ASSETS_DIR` - defaults to `.wordpress-org`, customizable for other locations of WordPress.org plugin repository-specific assets that belong in the top-level `assets` directory (the one on the same level as `trunk`).
 * `BUILD_DIR` - defaults to `false`. Set this flag to the directory where you build your plugins files into, then the action will copy and deploy files from that directory. Both absolute and relative paths are supported. The relative path if provided will be concatenated with the repository root directory. All files and folders in the build directory will be deployed, `.disignore` or `.gitattributes` will be ignored.
+* `DRY_RUN` - defaults to `false`. Set `true` to skip the final commit step and check all the debugging processes.
```

Comment:
Looking at this again, I think this would actually make more sense as a proper [input](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs) for the action, like [`generate-zip`](https://github.com/10up/action-wordpress-plugin-deploy/blob/f45e3dee4ea8b75a67f1b45b583fdb6a83166ff3/action.yml#L8-L10). This makes it part of the actual API of the action and less fragile to variance of however someone might use/pass an environment variable. Ultimately it would still be an environment variable in the script, but one that isn't intended to be passed manually.

Then we could follow the same simple conditional style that is used for `$INPUT_GENERATE_ZIP` https://github.com/10up/action-wordpress-plugin-deploy/blob/f45e3dee4ea8b75a67f1b45b583fdb6a83166ff3/deploy.sh#L163

What do you think?

Reply:
Thanks @aaemnnosttv, for the feedback. Do we need to add `dry-run.yml` to this PR so others can see how it will be used?

Reply:
I don't think we would need a new `dry-run.yml` file. I would suggest updating the [action.yml file](https://github.com/10up/action-wordpress-plugin-deploy/blob/develop/action.yml#L7) to set a default for the input

```
inputs:
  generate-zip:
    description: 'Generate package zip file?'
    default: false
  dry-run:
    description: 'Run the deployment process without committing.'
    default: false
```
And then set the environment variable based on the input [here](https://github.com/10up/action-wordpress-plugin-deploy/blob/develop/action.yml#L20)

```
      env:
        INPUT_GENERATE_ZIP: ${{ inputs.generate-zip }}
        INPUT_DRY_RUN: ${{ inputs.dry-run }}
```

Reply:
Thanks @joemcgill 👍

@mukeshpanchal27 this looks like the only part that's left to do here 🙂